protocol/msgpack/options.go

```diff
@@ -42,12 +45,77 @@ const (
 	defaultAggregatedReaderBufferSize = 1440
 )
 
+type baseEncoderOptions struct {
+	enabled    bool
+	compressor policy.Compressor
+}
+
+// NewBaseEncoderOptions creates a new set of base encoder options.
+func NewBaseEncoderOptions() BaseEncoderOptions {
+	return baseEncoderOptions{
+		compressor: policy.NewNoopCompressor(),
+	}
+}
+
+func (o baseEncoderOptions) SetPolicyCompressionEnabled(enabled bool) BaseEncoderOptions {
+	opts := o
```

Comment:
If we are passing the object by value, you don't need to make another copy of it. Can simply do
```
o.enabled = enabled
return o
```